dist/index.js

```diff
@@ -66,20 +68,29 @@ function run() {
             const labelGlobs = yield getLabelGlobs(client, configPath);
             const labels = [];
             const labelsToRemove = [];
+            const excessLabels = [];
             for (const [label, globs] of labelGlobs.entries()) {
                 core.debug(`processing ${label}`);
                 if (checkGlobs(changedFiles, globs)) {
-                    labels.push(label);
+                    if (labels.length >= GITHUB_MAX_LABELS) {
```

Comment:
I think this neglects the number of labels that are already present. Say the PR has 50 labels, and we come up with 100 that we want to apply, but only 25 of those are already present. The correct result would be to add 50 new ones, 25 remain the same, and 25 would be excess. But the current code reports 100 labels with no excess.

Reply:
Good point. I've moved the `addLabels` to happen AFTER the `removeLabels` (which happens with `syncLabels` enabled only) to mitigate this issue. But maybe there's a better way still, I'll check again. Thanks!

Reply:
To me, an easy solution seems to be first getting the labels already on the issue with [`listLabelsOnIssue`](https://octokit.github.io/rest.js/v19#issues-list-labels-on-issue). Then, you have all the information needed to ensure the _total_ count remains less than 100 while tracking accurately all additions/removals that you want to do.